libraries/AP_RCTelemetry/AP_CRSF_Telem.h

```diff
@@ -213,49 +246,76 @@ class AP_CRSF_Telem : public AP_RCTelemetry {
         BATTERY,
         GPS,
         FLIGHT_MODE,
+        PASSTHROUGH,
+        STATUS_TEXT,
         NUM_SENSORS
     };
 
     // passthrough WFQ scheduler
     bool is_packet_ready(uint8_t idx, bool queue_empty) override;
     void process_packet(uint8_t idx) override;
     void adjust_packet_weight(bool queue_empty) override;
-
+    void setup_custom_telemetry();
+    void update_custom_telemetry_rates(uint8_t rf_mode);
+    
     void calc_parameter_ping();
     void calc_heartbeat();
     void calc_battery();
     void calc_gps();
     void calc_attitude();
     void calc_flight_mode();
     void calc_device_info();
+    void calc_device_ping();
     void calc_parameter();
 #if HAL_CRSF_TELEM_TEXT_SELECTION_ENABLED
     void calc_text_selection( AP_OSD_ParamSetting* param, uint8_t chunk);
 #endif
     void update_params();
     void update_vtx_params();
+    void get_passthrough_telem_data();
+    void get_passthrough_telem_data_array();
+    void calc_status_text();
+    void process_rf_mode_changes();
+    uint8_t get_custom_telem_frame_id() const;
+    uint8_t get_rf_mode() const;
 
     void process_vtx_frame(VTXFrame* vtx);
     void process_vtx_telem_frame(VTXTelemetryFrame* vtx);
     void process_ping_frame(ParameterPingFrame* ping);
     void process_param_read_frame(ParameterSettingsReadFrame* read);
     void process_param_write_frame(ParameterSettingsWriteFrame* write);
+    void process_device_info_frame(ParameterDeviceInfoFrame* info);    
 
      // setup ready for passthrough operation
     void setup_wfq_scheduler(void) override;
-
+    
      // get next telemetry data for external consumers
     bool _get_telem_data(AP_RCProtocol_CRSF::Frame* data);
     bool _process_frame(AP_RCProtocol_CRSF::FrameType frame_type, void* data);
 
     TelemetryPayload _telem;
     uint8_t _telem_size;
     uint8_t _telem_type;
-
+    uint8_t _telem_rf_mode;
+    
     bool _telem_pending;
     bool _enable_telemetry;
-    uint8_t _request_pending;
 
+    struct {
+        uint8_t destination = AP_RCProtocol_CRSF::CRSF_ADDRESS_BROADCAST;
+        uint8_t frame_type;
+    } PACKED _request_pending;
+
+    struct {
+        uint8_t minor;
+        uint8_t major;
+        uint8_t retry_count;
+        bool use_rf_mode;
+        bool pending = true;
```

Comment:
Should this really be pending as a default?

Reply:
yes, since at boot I need to get the version to complete custom telemetry initialization.
I could save some by not initializing it here but would need to set it to true somewhere during initialization, no idea about the impact.